src/master/master.js

```diff
@@ -119,21 +108,19 @@ export class Master {
    */
   async onUpdate(action, stateID, gameID, playerID) {
     let isActionAuthentic;
-
+    const { credentials } = action.payload || {};
     if (this.executeSynchronously) {
       const gameMetadata = this.storageAPI.get(GameMetadataKey(gameID));
-      isActionAuthentic = this.auth({
-        action,
-        gameMetadata,
-        playerID,
-      });
+      const playerMetadata = getPlayerMetadata(gameMetadata, playerID);
+      isActionAuthentic = this.shouldAuth(gameMetadata)
```

Comment:
Not super critical to test, but in the interest of keeping the test coverage at 100%, can we add a test to cover both branches here?  If you run `npm run test:coverage` on the repo, you will notice that it complains about uncovered branches at line 115 here and line 66 in `index.ts`. I'm happy to also follow up with these tests if you don't have time to add them here.

Reply:
I’m just realising that the `executeSynchronously` branches aren’t being tested at all in the `Master` tests, because they’re only used by `LocalMaster` in the local transport implementation. Because there’s no game metadata in the `LocalMaster` tests, the `shouldAuth` branch always returns false. I don’t think I’m familiar enough with the local transport to fix this.

Seeing this though, I’ve tweaked the way `shouldAuth` is initialised, because previously I had it calling `doesGameRequireAuthentication` even if authentication hadn’t been enabled.

Reply:
Actually, it’s not just that there’s no metadata, `LocalMaster` creates an instance of `Master` that has auth disabled. So there’s never currently a `Master` that executes synchronously _and_ uses authentication.